quickwit-search/src/fetch_docs.rs

```diff
@@ -144,30 +163,140 @@ async fn get_searcher_for_split_without_cache(
     Ok(reader)
 }
 
+/// A struct for holding a fetched document's content and highlights.
+struct DocBundle {
+    doc_json: String,
+    highlights_json: Option<String>,
+}
+
 /// Fetching docs from a specific split.
 #[tracing::instrument(skip(global_doc_addrs, index_storage, split))]
 #[allow(clippy::needless_lifetimes)]
 async fn fetch_docs_in_split(
     mut global_doc_addrs: Vec<GlobalDocAddress>,
     index_storage: Arc<dyn Storage>,
     split: &SplitIdAndFooterOffsets,
-) -> anyhow::Result<Vec<(GlobalDocAddress, String)>> {
+    doc_mapper: Arc<dyn DocMapper>,
+    search_request: &SearchRequest,
+) -> anyhow::Result<Vec<(GlobalDocAddress, DocBundle)>> {
     global_doc_addrs.sort_by_key(|doc| doc.doc_addr);
 
     let index_reader = get_searcher_for_split_without_cache(index_storage, split).await?;
-    let searcher = Arc::new(index_reader.searcher());
+    let searcher = index_reader.searcher();
+    let snippet_generators =
+        Arc::new(create_snippet_generators(&searcher, doc_mapper, search_request).await?);
+
     let doc_futures = global_doc_addrs.into_iter().map(|global_doc_addr| {
         let searcher = searcher.clone();
+        let moved_snippet_generators = snippet_generators.clone();
```

Comment:
Generally we use `_clone` when we clone. `moved` in the name suggest it's already moved but it's not the case at this line :)